src/pages/ReferralPage.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ReactElement, useEffect, useState } from "react";
+import Image from "react-bootstrap/Image";
+import Button from "react-bootstrap/Button";
+import UserBanner from "src/assets/HowFree/UsersBanner.png";
+import "./ReferralPage.css";
+import LottieLetter from "src/assets/Referral/LottieLetter.json";
+import Lottie from "react-lottie";
+import CardContentLoader from "src/components/loading/CardContentLoader";
+import GenericModal from "src/components/modals/GenericModal";
+
+import { RouteComponentProps, useHistory } from "react-router-dom";
+import { fetchReferrerName } from "src/services/Api/index";
+import { trackButtonClick } from "src/utils/analytics";
+import { BUTTON_TYPES, PLACEMENT } from "src/utils/constants";
+
+type TParams = { id: string };
+
+export default function Referral({
+  match,
+}: RouteComponentProps<TParams>): ReactElement {
+  const [referrer, setReferrer] = useState<string>("");
```

Comment:
idk if you need the `<type>` on these, I could be mistaken but for me it's usually inferred

Reply:
think it's good to be explicit!
but ye, typescript don't enforce it